### PR TITLE
Upgrade all dependencies

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt [hunk 1]
```diff
@@ -38,31 +38,31 @@ class CodeBlockTest {
   @Test fun indentCannotBeIndexed() {
     assertThrows<IllegalArgumentException> {
       CodeBlock.builder().add("%1>", "taco").build()
-    }.hasMessage("%%, %>, %<, %[, %], and %W may not have an index")
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt]: `hasMessage()` is deprecated

File: src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt [hunk 2]
```diff
@@ -38,31 +38,31 @@ class CodeBlockTest {
...
+    }.hasMessageThat().isEqualTo("%%, %>, %<, %[, %], and %W may not have an index")
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt]: `hasMessage()` is deprecated

File: pom.xml [hunk 3]
```diff
@@ -86,19 +86,19 @@
...
     </dependency>
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
-      <version>1.10.16</version>
+      <version>2.11.0</version>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.eclipse.jdt.core.compiler</groupId>
```
Comment (JakeWharton) [inline pom.xml]: Is this still used?

Comment (Egorand) [reply pom.xml]: Not sure, can check

Comment (Egorand) [reply pom.xml]: There's a single test in `FunSpecTest` that uses Mockito, maybe there's a way to rewrite it and get rid of the dependency

**Pull request merged**